homeassistant/components/lg_netcast/config_flow.py

```diff
@@ -0,0 +1,213 @@
+    DOMAIN,
+)
+from .helpers import LGNetCastDetailDiscoveryError, async_discover_netcast_details
+
+
+class LGNetCast(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for LG Netcast TV integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize config flow."""
+        self.client: LgNetCastClient | None = None
+        self.device_config: dict[str, Any] = {}
+        self._discovered_devices: dict[str, Any] = {}
+        self._track_interval: CALLBACK_TYPE | None = None
+
+    def create_client(self) -> None:
+        """Create LG Netcast client from config."""
+        host = self.device_config[CONF_HOST]
+        access_token = self.device_config.get(CONF_ACCESS_TOKEN)
+        self.client = LgNetCastClient(host, access_token)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            if is_host_valid(host):
+                self.device_config[CONF_HOST] = host
+                return await self.async_step_authorize()
+
+            errors[CONF_HOST] = "invalid_host"
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema({vol.Required(CONF_HOST): str}),
+            errors=errors,
+        )
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Import configuration from yaml."""
+        self.device_config = {
+            CONF_HOST: config[CONF_HOST],
+            CONF_NAME: config[CONF_NAME],
+        }
+        await self.async_discover_client()
+
+        try:
+            self._async_abort_entries_match({CONF_ID: self.device_config[CONF_ID]})
+        except AbortFlow as err:
+            async_create_issue(
+                self.hass,
+                DOMAIN,
+                "deprecated_yaml_import_issue_already_configured",
+                breaks_in_ha_version="2024.6.0",
+                is_fixable=False,
+                issue_domain=DOMAIN,
+                severity=IssueSeverity.WARNING,
+                translation_key="deprecated_yaml_import_issue_already_configured",
+                translation_placeholders={
+                    "domain": DOMAIN,
+                    "interation_title": "LG Netcast",
+                },
+            )
+            raise err
+
+        async_create_issue(
+            self.hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2024.6.0",
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "LG Netcast",
+            },
+        )
```

Comment:
The way in which we distinguish between these two cases is not correct, because it makes no difference to the user.
Instead, we should distinguish between:
1. The import is done, and the user should remove YAML
2. Something went wrong when doing import (if this can happen, for example in the authorize step, and we can't leave the flow import for the user to finish)

Reply:
I've refactored this and now raising issues if we get an error - and only raising the deprecated_yaml issue if we succeed (or fail because it's already migrated)